truck_delivery_yess/db_connector.py

```diff
@@ -1,11 +1,71 @@
 import redis
+import json
 
 
 class DBConnector:
+    __instance = None
+    __connection = None
+
     def __init__(self):
-        pass
+        if DBConnector.__instance is not None:
+            raise Exception("This class is a singleton!")
+        else:
+            DBConnector.__instance = self
+            self.__connection = redis.Redis(host="localhost", port=6379)
```

Comment:
This implementation is fine, as you mentioned on class this implementation can be done in any language, which is also described here https://python-patterns.guide/gang-of-four/singleton/#the-gang-of-fours-implementation

But as you can see in a section below a more python implementation is described https://python-patterns.guide/gang-of-four/singleton/#a-more-pythonic-implementation